File: 준영/프로그래머스/LV2/괄호 회전하기.js
function solution(s) {
  var answer = 0
  var temp = s
  for (let i = 0; i < s.length; i++) {
    const my_stack = []
    var is_right = true
    for (const j of temp) {
      if (j === '[' || j === '{' || j === '(') {
        my_stack.push(j)
      } else if (my_stack.length === 0) {
        is_right = false
        break
      }

      if (j == ']') {
        if (my_stack[my_stack.length-1] === '[') {
          my_stack.pop()
        } else {
          is_right = false
          break
        }
      }

      if (j == '}') {
        if (my_stack[my_stack.length-1] === '{') {
          my_stack.pop()
        } else {
          is_right = false
          break
        }
      }

      if (j == ')') {
        if (my_stack[my_stack.length-1] === '(') {
          my_stack.pop()
        } else {
          is_right = false
          break
        }
      }
    }
    if (my_stack.length !== 0) {
      is_right = false
    }

    if (is_right) {
      answer += 1
    }
    var x = temp[0]
    temp = temp.substring(1) + x
  }
  return answer
}

File: 준영/프로그래머스/LV2/괄호 회전하기.py
def solution(s):
    answer = 0
    temp = s
    for i in range(len(s)):
        my_stack = []
        is_right = True
        for j in temp:
            if j == '[' or j == '{' or j == '(':
                my_stack.append(j)
            
            if j == ']': 
                if len(my_stack) == 0:
                    is_right = False
                    break
                if my_stack[-1] == '[':
                    my_stack.pop()
                else:
                    is_right = False
            
            if j == '}':
                if len(my_stack) == 0:
                    is_right = False
                    break
                if my_stack[-1] == '{':
                    my_stack.pop()
                else:
                    is_right = False
            
            if j == ')':
                if len(my_stack) == 0:
                    is_right = False
                    break
                if my_stack[-1] == '(':
                    my_stack.pop()
                else:
                    is_right = False
        if len(my_stack):
            is_right = False
            
        if is_right:
            answer += 1

        temp = temp[1:] + temp[0]
    
    return answer

s = "[](){}"	
solution(s)